File: ZMP/clock_angle.py
def get_angle(hour, minute): # hour: 0~23, minute: 0~59
    min_angle= minute / 60 * 360
    if hour >= 12:
        hour= hour - 12
    hour_angle= hour / 12 * 360 + min_angle / 12
    angle= min_angle - hour_angle
    if angle < 0:
        return angle * -1
    return angle

print("00:15 ->", get_angle(0, 15))
print("01:59 ->", get_angle(1, 59))
print("07:26 ->", get_angle(7, 26))
print("19:19 ->", get_angle(19, 19))
print("22:07 ->", get_angle(22, 7))
print("11:00 ->", get_angle(11, 00))
print("14:30 ->", get_angle(14, 30))

File: ZMP/PhamVanNam/codes/clock_angle.cpp
#include <iostream>

using namespace std;

float getAngle(int hour, int minute){
    float min_angle= float(minute) / 60 * 360;
    if (hour >= 12){
        hour= hour - 12;
    }        
    float hour_angle= float(hour) / 12 * 360 + min_angle / 12;
    float angle= min_angle - hour_angle;
    
    if (angle < 0){
        return angle * -1;
    }
    return angle;
}

int main() {
    printf("00:15 -> %f\n", getAngle(0, 15));
    printf("01:59 -> %f\n", getAngle(1, 59));
    printf("07:26 -> %f\n", getAngle(7, 26));
    printf("19:19 -> %f\n", getAngle(19, 19));
    printf("22:07 -> %f\n", getAngle(22, 7));
    printf("11:00 -> %f\n", getAngle(11, 00));
    printf("14:30 -> %f\n", getAngle(14, 30));
    return 0;
}